### PR TITLE
Add secret file resource

### PR DESCRIPTION
**Pull request by rathodmeetsatish**
Add secret file resource

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -92,4 +92,4 @@ require (
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 )
 
-// replace github.com/harness/harness-go-sdk => ../harness-go-sdk
+replace github.com/harness/harness-go-sdk => /Users/meetrathod/harness-go-sdk
```
Comment (ashishsanodia-harness) [inline go.mod:95]: This does not seem correct

Comment (micahlmartin) [reply go.mod:95]: This needs to be removed. It's for debugging only

File: internal/service/platform/secret/file.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+			OrgIdentifier:     buildField(d, "org_id"),
+			ProjectIdentifier: buildField(d, "project_id"),
+			Spec:              optional.NewString(fmt.Sprintf(`{"secret":{"type":"SecretFile","name":"%[1]s","identifier":"%[2]s","description":"%[3]s","tags":%[4]s,"spec":{"secretManagerIdentifier":"%[5]s"}}}`, d.Get("name"), d.Get("identifier"), d.Get("description"), strings.Join(tags_string, ","), d.Get("secret_manager_identifier"))),
+			File:              optional.NewInterface(d.Get("file_path").(string)),
+		})
+	} else {
+		resp, _, err = c.SecretsApi.PutSecretFileV2(ctx, c.AccountId, id, &nextgen.SecretsApiPutSecretFileV2Opts{
+			OrgIdentifier:     buildField(d, "org_id"),
+			ProjectIdentifier: buildField(d, "project_id"),
+			Spec:              optional.NewString(fmt.Sprintf(`{"secret":{"type":"SecretFile","name":"%[1]s","identifier":"%[2]s","description":"%[3]s","tags":%[4]s,"spec":{"secretManagerIdentifier":"%[5]s"}}}`, d.Get("name"), d.Get("identifier"), d.Get("description"), strings.Join(tags_string, ","), d.Get("secret_manager_identifier"))),
```
Comment (micahlmartin) [inline internal/service/platform/secret/file.go:80]: This seems wrong to me. https://harness.io/docs/api/tag/Secrets#operation/putSecretFileV2. What is `spec` supposed to be here? In the API it's just a string which seems incorrect.

File: internal/service/platform/secret/file.go [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+	}
+
+	if err := readSecretFile(d, resp.Data.Secret); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return nil
+}
+
+func buildTag(no_of_tags int, tags *schema.Set) []string {
```
Comment (micahlmartin) [inline internal/service/platform/secret/file.go:96]: You should probably be able use `helpers.ExpandTags` and `helpers.FlattenTags` for this.

File: internal/service/platform/secret/file_test.go [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+func testAccResourceSecret_file(id string, name string) string {
+	return fmt.Sprintf(`
+	resource "harness_platform_secret_file" "test" {
+		identifier = "%[1]s"
+		name = "%[2]s"
+		description = "test"
+		tags = ["foo:bar"]
+		file_path = "/Users/meetrathod/terraform-provider-harness/internal/acctest/secret_files/secret.txt"
```
Comment (micahlmartin) [inline internal/service/platform/secret/file_test.go:65]: This should be a path to a file that lives somewhere in like `acctest/files`

File: internal/service/platform/secret/file.go (hunk 3 above)
Comment (rathodmeetsatish) [reply internal/service/platform/secret/file.go:96]: As we have to send spec in string form so wanted tags in a spectic format so had to create a new method

**Review by rijajoo — APPROVED** [review]

**Pull request merged**